### PR TITLE
make formats docs folder, create COCO format documentation

### PR DESCRIPTION
**Pull request by zmaslova**
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/formats/coco_user_manual.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+# converting to COCO format from other format
+datum convert -if voc -i <path/to/voc/dataset> \
+    -f coco -o <path/to/export/dir> -- --save-images
+```
+
+Extra options for export to COCO format:
+- `--save-images` allow to export dataset with saving images
+(by default `False`);
+- `--image-ext IMAGE_EXT` allow to specify image extension
+for exporting dataset (by default `.png`);
```
Comment (zhiltsov-max) [inline docs/formats/coco_user_manual.md:150]:
```suggestion
for exporting dataset (by default - keep original or use `.jpg`, if none);
```

File: docs/formats/coco_user_manual.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+- `--segmentation-mode MODE` allow to specify save mode for instance segmentation:
+    - 'guess': guess the mode for each instance (using 'is_crowd' attribute as hint)
+    - 'polygons': save polygons( merge and convert masks, prefer polygons)
+    - 'mask': save masks (merge and convert polygons, prefer masks)
+(by default `guess`);
+- `--crop-covered` allow torop covered segments so that background objects
```
Comment (zhiltsov-max) [inline docs/formats/coco_user_manual.md:156]:
```suggestion
- `--crop-covered` allow to crop covered segments so that background objects
```

**Review by zhiltsov-max — APPROVED** [review]

File: docs/formats/coco_user_manual.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+by default Datumaro uses all tasks. Example:
+
+```bash
+datum import -o project -f coco -i <dataset>
+datum export -p project -f coco -- --tasks instances,stuff
+```
+
+## Particular use cases
+
+Datumaro supports filtering, transformation, merging etc. for all formats
```
Comment (zhiltsov-max) [inline docs/formats/coco_user_manual.md:174]: Add 2 examples: how to load the original format and how to create a custom dataset in this format. You can also add a reference to tests.

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

File: docs/formats/coco_user_manual.md (hunk 1 above)
Comment (zmaslova) [reply docs/formats/coco_user_manual.md]: Fixed

File: docs/formats/coco_user_manual.md (hunk 3 above)
Comment (zmaslova) [reply docs/formats/coco_user_manual.md:174]: Is it enough?

File: docs/formats/coco_user_manual.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+``` bash
+datum import --format coco_stuff --input-path <path/to/stuff.json>
+```
+
+Datumaro supports the following COCO tasks:
+- [Image Captioning](https://cocodataset.org/#captions-2015) (`coco_caption`)
+- [Object Detection](https://cocodataset.org/#detection-2020) (`coco_instances`, `coco_label`)
+- [Panoptic Segmentation](https://cocodataset.org/#panoptic-2020) (`coco_panoptic`)
+- [Keypoint Detection](https://cocodataset.org/#keypoints-2020) (`coco_person_keypoints`)
+- [Stuff Segmentation](https://cocodataset.org/#stuff-2019) (`coco_stuff`)
```
Comment (zhiltsov-max) [inline docs/formats/coco_user_manual.md:108]:
```suggestion
- [Image Captioning](https://cocodataset.org/#captions-2015) (`coco_caption`)
- [Object Detection](https://cocodataset.org/#detection-2020) (`coco_instances`)
- Image classification (our extension) (`coco_labels`) - a format like Object Detection, which uses 
  only `category_id` and `score` annotation fields
- [Panoptic Segmentation](https://cocodataset.org/#panoptic-2020) (`coco_panoptic`)
- [Keypoint Detection](https://cocodataset.org/#keypoints-2020) (`coco_person_keypoints`)
- [Stuff Segmentation](https://cocodataset.org/#stuff-2019) (`coco_stuff`)
```

File: docs/formats/coco_user_manual.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                        id=20, group=20, attributes={'is_crowd': True}),
+                ]
+            ),
+    ], categories=['a', 'b', 'c', 'd'])
+
+dataset.export('./dataset', format='coco_panoptic')
+```
+
+Some examples of working with Pascal VOC dataset from code you can found in
+[tests](../../tests/test_voc_format.py)
```
Comment (zhiltsov-max) [inline docs/formats/coco_user_manual.md:216]:
```suggestion
More examples of working with COCO dataset from code can be found in
[tests](../../tests/test_coco_format.py)
```

File: docs/formats/mnist_user_manual.md [hunk 6]
```diff
@@ -174,3 +174,6 @@ datum convert --input-format mnist --input-path <path/to/mnist> \
               --output-format imagenet \
               --filter '/item[annotation/label="3"]'
 ```
+
+Some examples of working with Pascal VOC dataset from code you can found in
```
Comment (zhiltsov-max) [inline docs/formats/mnist_user_manual.md:178]:
```suggestion
More examples of working with MNIST dataset from code can be found in
```

File: docs/formats/pascal_voc_user_manual.md [hunk 7]
```diff
@@ -314,4 +314,4 @@ in Pascal VOC 2012 has for segmentation task and this result is the same as the
 [official documentation](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/dbstats.html)
 
 Some examples of working with Pascal VOC dataset from code you can found in
```
Comment (zhiltsov-max) [inline docs/formats/pascal_voc_user_manual.md:316]:
```suggestion
More examples of working with Pascal VOC dataset from code can be found in
```

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

**Pull request merged**